### PR TITLE
Change database yml directly and double down on separate DB

### PR DESCRIPTION
**Pull request by dhh**
For Solid Cache to become a Rails 8 default, the configuration needs to be fully automated, such that we can run it as part of "rails new". This PR makes the installation a single step process, and commits to using a separate database for the default case.

This also converts the setup from using a migration to providing the full SC db as a schema. This is needed in order to only have the cache db configured in production. Otherwise you also need it configured in development, even if you're not using it, or the migration installation won't run. It's better anyway like this.

### CODE REVIEW

File: lib/generators/solid_cache/install/install_generator.rb [hunk 1]
```diff
@@ -16,9 +16,65 @@ def create_config_solid_cache_yml
     template "config/solid_cache.yml"
   end
 
+  def add_cache_db_to_database_yml
+    if app_root.join("config/database.yml").exist?
+      if app_is_using_sqlite?
+        gsub_file database_yml, /production:\s*<<: \*default.*/m, sqlite_database_config_with_cache
```
Comment (RenatoRosaFranco) [inline lib/generators/solid_cache/install/install_generator.rb:22]: @dhh what you think if we removed nested if statements?

```
def add_cache_db_to_database_yml
  return unless app_root.join("config/database.yml").exist?

  if app_is_using_sqlite?
    update_database_yml_with_cache(sqlite_database_config_with_cache)
  else
    update_database_yml_with_cache(generic_database_config_with_cache)
  end
end

private

def update_database_yml_with_cache(config)
  gsub_file database_yml, /production:\s*<<: \*default.*/m, config
end
```

File: lib/generators/solid_cache/install/install_generator.rb [hunk 2]
```diff
@@ -7,7 +7,7 @@ class SolidCache::InstallGenerator < Rails::Generators::Base
                                     desc: "Skip migrations"
 
   def add_rails_cache
-    if (env_config = Pathname(destination_root).join("config/environments/production.rb")).exist?
+    if (env_config = app_root.join("config/environments/production.rb")).exist?
```
Comment (RenatoRosaFranco) [inline lib/generators/solid_cache/install/install_generator.rb:10]: @dhh  suggestion.

```
def add_rails_cache
  production_config_file = app_root.join("config/environments/production.rb")

  if production_config_file.exist?
    configure_cache_store(production_config_file)
  end
end

private

def configure_cache_store(config_file)
  gsub_file config_file, /(# )?config\.cache_store = (:.*)/, "config.cache_store = :solid_cache_store"
end
```

File: lib/generators/solid_cache/install/install_generator.rb (hunk 1 above)
Comment (dhh) [reply lib/generators/solid_cache/install/install_generator.rb:19]: I actually want to get rid of these defensive clauses entirely. They're only used for testing. It's fine for the installer to explode if you don't have config/database.yml.

File: lib/generators/solid_cache/install/install_generator.rb (hunk 2 above)
Comment (dhh) [reply lib/generators/solid_cache/install/install_generator.rb]: Got rid of the needlessly defensive guard instead.

File: lib/generators/solid_cache/install/install_generator.rb [hunk 3]
```diff
@@ -3,22 +3,70 @@
...
   end
 
-  def create_migrations
-    unless options[:skip_migrations]
-      rails_command "railties:install:migrations FROM=solid_cache", inline: true
+  def add_cache_db_to_database_yml
+    if app_is_using_sqlite?
+      gsub_file database_yml, /production:\s*<<: \*default.*/m, sqlite_database_config_with_cache
+    else
+      gsub_file database_yml, /production:\s*<<: \*default.*/m, generic_database_config_with_cache
```
Comment (fractaledmind) [inline lib/generators/solid_cache/install/install_generator.rb:19]: Not sure how you want to handle this situation, but if someone is adding Solid Cache to an application where they have already added, for example, Solid Queue—and thus their `database.yml` is already 3-tier, this won't add the cache database to their config.

Comment (dhh) [reply lib/generators/solid_cache/install/install_generator.rb:19]: Yes, this needs to be more resilient. If you're interested, please do take a look to make it so! We want Solid Cache and Solid Queue configured out of the box and with sqlite by default, so this does need to be resolved.

Comment (dhh) [reply lib/generators/solid_cache/install/install_generator.rb:19]: Going to release with this for now to move the ball forward on integration. But we will have to revisit this again when we have a fix so we can get solid queue in on the same deal.

**Pull request merged**